model/task/task.go

```diff
@@ -1679,20 +1672,61 @@ func (t *Task) CircularDependencies() error {
 	return catcher.Resolve()
 }
 
-func (t *Task) IsBlockedDisplayTask() bool {
-	if !t.DisplayOnly {
-		return false
+func (t *Task) FindAllUnmarkedBlockedDependencies(blocked bool) ([]Task, error) {
+	okStatusSet := []string{AnyStatus}
+	if !blocked {
+		okStatusSet = append(okStatusSet, t.Status, AllStatuses)
 	}
+	query := db.Query(bson.M{
+		DependsOnKey: bson.M{"$elemMatch": bson.M{
+			DependencyTaskIdKey:       t.Id,
+			DependencyStatusKey:       bson.M{"$nin": okStatusSet},
+			DependencyUnattainableKey: false,
+		},
+		}})
 
-	tasksWithDeps, err := FindAllTasksFromVersionWithDependencies(t.Version)
+	return FindAll(query)
+}
+
+func (t *Task) UpdateBlockedDependencies(blocked bool) error {
```

Comment:
This is never called with `blocked == true`

Reply:
It's called recursively with blocked == true
(https://github.com/evergreen-ci/evergreen/pull/2344/files#diff-ec70289495c0b3b32aa040cc657874cfR1701)

Reply:
Thanks.